### PR TITLE
Corrected Spore Explosion and Earth Shaker skills

### PR DESCRIPTION
**Pull request by Haydrich**
* **Addressed Issue(s)**: Part of #8378 

* **Server Mode**: RE

* **Description of Pull Request**: I was watching iRO videos and found that Spore Explosion bonus is missing an effect
https://youtu.be/9ELfg4PtCg4?si=7vVXrwkQqY44tzWI&t=60
and about the Earth Shaker i tested it in kRO, it should give a Red Effect on target and the bonus only works with mobs, not players.
https://www.divine-pride.net/forum/index.php?/topic/3723-kro-jobs-improvement-project/#comment-5519
image

### CODE REVIEW

File: src/map/status.cpp [hunk 1]
```diff
@@ -13714,6 +13714,9 @@ int status_change_end(struct block_list* bl, enum sc_type type, int tid)
 		case SC_INTRAVISION:
 			calc_flag = status_db.getSCB_ALL(); // Required for overlapping
 			break;
+        case SC_EARTHSHAKER:
```
Comment (AoShinRO) [inline src/map/status.cpp:13717]: can you adjust the indentation here?

File: src/map/skill.cpp [hunk 2]
```diff
@@ -1956,8 +1956,11 @@ int skill_additional_effect( struct block_list* src, struct block_list *bl, uint
 		sc_start(src,bl, SC_SILENCE, 5 * skill_lv + (status_get_dex(src) + status_get_lv(src)) / 10, skill_lv, skill_get_time(skill_id, skill_lv));
 		break;
 	case SR_EARTHSHAKER:
+		if( bl->type == BL_MOB ) {
+			sc_start(src, bl, SC_EARTHSHAKER, 100, skill_lv, skill_get_time2(skill_id, skill_lv));
+			clif_specialeffect(bl, 1398, AREA);
```
Comment (Atemo) [inline src/map/skill.cpp:1961]: Maybe it's an efst rather than an effect?

**Review by Atemo — APPROVED** [review]
It looks perfect

**Review by Lemongrass3110 — APPROVED** [review]

File: src/map/skill.cpp [hunk 3]
```diff
@@ -1956,8 +1956,9 @@ int skill_additional_effect( struct block_list* src, struct block_list *bl, uint
 		sc_start(src,bl, SC_SILENCE, 5 * skill_lv + (status_get_dex(src) + status_get_lv(src)) / 10, skill_lv, skill_get_time(skill_id, skill_lv));
 		break;
 	case SR_EARTHSHAKER:
+		if( bl->type == BL_MOB )
```
Comment (Atemo) [inline src/map/skill.cpp:1959]: To exclude guardian type
```suggestion
		if (dstmd != nullptr && dstmd->guardian_data == nullptr)    // Target is a mob (boss included) and not a guardian type
```

**Pull request merged**